Server/mods/deathmatch/logic/CAPI.cpp

```diff
@@ -0,0 +1,81 @@
+#include "StdInc.h"
+#include "CAPI.h"
+#include "CGame.h"
+#include "CMainConfig.h"
+#include <SharedUtil.String.h>
+
+namespace Endpoints
+{
+    inline const std::string API_ENDPOINT = "/api";
+    namespace Server
+    {
+        inline const std::string PLAYERS = "/players";
+        inline const std::string MAX_PLAYERS = PLAYERS + "/max";
+    }
+}
+
+const bool GetNextPathFragment(std::string& path) noexcept
+{
+    auto index = path.find('/', 1);
+    if (index == path.npos)
+        return false;
+
+    path = path.substr(index);
+    return true;
+};
+
+ResponseCode CAPI::HandleRequest(HttpRequest* request, HttpResponse* response)
+{
+    response->oResponseHeaders["content-type"] = "application/json";
+
+    std::string uri = request->sOriginalUri;    
+
+    if (!SharedUtil::StartsWith(uri, Endpoints::API_ENDPOINT))
+        return HTTPRESPONSECODE_404_NOTFOUND;
+
+    if (!GetNextPathFragment(uri))
+        return HTTPRESPONSECODE_404_NOTFOUND;
+
+    if (SharedUtil::StartsWith(uri, Endpoints::Server::PLAYERS))
+    {
+        return HandlePlayers(request, response);
+    }
+
+    std::string body("{\"error\": \"404: Not Found\"}");
+    response->SetBody(body.c_str(), body.size());
```

Comment:
Can't a C-style string be passed in directly?
I don't see why we must use an `std::string`.

Reply:
> Can't a C-style string be passed in directly? I don't see why we must use an `std::string`.

Only for simplicity

Reply:
Make at least it `static`

Reply:
Or even better std::string_view